### PR TITLE
New sign up page with form completed and major and year dropdown done.

### PR DESCRIPTION
**Pull request by thaigillespie**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+
+  <head>
+    <meta charset="utf-8">
+    <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <meta name="theme-color" content="#000000">
+    <!--
       manifest.json provides metadata used when your web app is added to the
       homescreen on Android. See https://developers.google.com/web/fundamentals/web-app-manifest/
     -->
```
Comment (godwinpang) [inline public/index.html:12]: why is this file changed?

File: src/components/Dropdowns/YearDropdown/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * as the default props for that <Field /> (field, form, meta) and transforms
+ * all that into a props that is suitable for MUI's <TextField />
+ */
+export function fieldToTextField({
+  disabled,
+  field,
+  form: { isSubmitting, touched, errors },
+  ...props
+}) {
+  const fieldError = getIn(errors, field.name);
```
Comment (godwinpang) [inline src/components/Dropdowns/YearDropdown/index.js:27]: what's getIn?

File: src/components/Dropdowns/YearDropdown/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  const showError = getIn(touched, field.name) && !!fieldError;
```
Comment (godwinpang) [inline src/components/Dropdowns/YearDropdown/index.js:28]: no bang bang hacks :) just check if it's null or sth idk what getIn returns

File: src/components/Dropdowns/YearDropdown/index.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+  return {
+    ...props,
+    ...field,
+    error: showError,
+    helperText: showError ? fieldError : props.helperText,
+    disabled: disabled ?? isSubmitting,
```
Comment (godwinpang) [inline src/components/Dropdowns/YearDropdown/index.js:35]: again change to null check for future sustainability - not a big fan of ??

File: src/components/EventEditForm/edit_form.js [hunk 5]
```diff
@@ -163,9 +163,21 @@ const EventEditForm = props => {
...
+    name: PropTypes.string.isRequired,
+    tags: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
+    description: PropTypes.string.isRequired,
+    urls: PropTypes.shape({
+      fb: PropTypes.string.isRequired,
+      canva: PropTypes.string.isRequired,
+      rsvp: PropTypes.string.isRequired,
+      signin: PropTypes.string.isRequired,
+    }),
+  }).isRequired,
```
Comment (godwinpang) [inline src/components/EventEditForm/edit_form.js:180]: lol looking at all these long af proptypes makes me miss typescript

File: src/components/Loading/index.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { LinearProgress as CircularProgress } from '@material-ui/core';
 
 const styles = theme => ({
   progress: {
-    margin: theme.spacing.unit * 3,
+    margin: theme.spacing(3),
```
Comment (godwinpang) [inline src/components/Loading/index.js:7]: ty for fixing

File: src/components/SignUpForm/index.js [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+
+import styles from './styles';
+import MajorDropdown from '../Dropdowns/MajorDropdown';
+import YearDropdown from '../Dropdowns/YearDropdown';
+
+const MIN_GRAD_YEAR = 2005;
+const MAX_GRAD_YEAR = moment().year() + 4;
+const PW_MIN_LENGTH = 4;
+
+const schema = Yup.object({
```
Comment (godwinpang) [inline src/components/SignUpForm/index.js:20]: consider putting schema in separate file for readability

File: src/components/SignUpForm/index.js [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+import { Button, Grid, LinearProgress } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { TextField } from 'formik-material-ui';
+
+import { Formik, Field, Form } from 'formik';
+import * as Yup from 'yup';
+
+import styles from './styles';
+import MajorDropdown from '../Dropdowns/MajorDropdown';
```
Comment (godwinpang) [inline src/components/SignUpForm/index.js:13]: add index.js in Dropdowns to export both so we only need 1 import statement
also change Dropdowns to lowercase to adhere with naming conventions throughout codebase

File: src/components/SignUpForm/index.js [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+                  minyear={MIN_GRAD_YEAR}
+                  maxyear={MAX_GRAD_YEAR}
+                  fullWidth
+                  name='gradYear'
+                  label='Grad Year'
+                />
+              </Grid>
+            </Grid>
+
+            {isSubmitting && <LinearProgress />}
```
Comment (godwinpang) [inline src/components/SignUpForm/index.js:140]: good

File: src/components/SignUpForm/styles.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+const styles = () => ({
+  nameFields: {
+    marginBottom: '8px',
```
Comment (godwinpang) [inline src/components/SignUpForm/styles.js:3]: in the scenarios that I can think of, we shouldn't ever ever have to hardcode any margin in terms of px

File: src/pages/SignUpPageNew/styles.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    marginTop: '28px',
+    marginBottom: '40px',
+  },
+  redirectToSignIn: {
+    marginTop: '20px',
+    color: '#5B5B5B',
+    textTransform: 'none',
+  },
+});
+
```
Comment (godwinpang) [inline src/pages/SignUpPageNew/styles.js:28]: again, dont see need to hardcode margin/paddings - let's meet about this sometime

Comment (godwinpang) [reply src/pages/SignUpPageNew/styles.js:28]: I could be missing something though - mainly just not sure about the hardcoded width (overflow small screens) and the margin/paddings

File: src/components/Dropdowns/YearDropdown/index.js (hunk 4 above)
Comment (thaigillespie) [reply src/components/Dropdowns/YearDropdown/index.js:35]: Honestly that stuff is the stuff that came from formik-material-ui that I didn't touch. I'll change it tho

File: public/index.html (hunk 1 above)
Comment (thaigillespie) [reply public/index.html:12]: I'm guessing indentation? Not sure why this was changed, it might have been done by VSCode

File: src/components/SignUpForm/index.js (hunk 8 above)
Comment (thaigillespie) [reply src/components/SignUpForm/index.js:13]: Wait but I thought the convention for src/components' naming of folders is capitalization of first letter of all words in a name? That's what I see for other folders. Am I missing something here?

File: src/pages/SignUpPageNew/styles.js (hunk 11 above)
Comment (thaigillespie) [reply src/pages/SignUpPageNew/styles.js:28]: The hardcoded width is for the major dropdown box messing everything up when a major name was longer than the actual box, tho it probably shouldn't matter now because all major names are shorter than the dropdown box now (as far as I can tell).

File: src/components/SignUpForm/index.js (hunk 8 above)
Comment (godwinpang) [reply src/components/SignUpForm/index.js:13]: yes that's true - and perhaps I shouldn't have said naming conventions throughout codebase.

My rationale was more along the lines of React codebases having only first letter capitalized for components - since dropdowns is a **folder** of components it shouldnt be capitalized

Comment (thaigillespie) [reply src/components/SignUpForm/index.js:13]: Oh I see now. Thank you so much for clarifying! Will change it accordingly

File: src/components/Dropdowns/YearDropdown/index.js (hunk 2 above)
Comment (thaigillespie) [reply src/components/Dropdowns/YearDropdown/index.js:27]: https://stackoverflow.com/questions/56089054/how-to-dynamically-access-nested-errors-touched-on-formik-field

This is probably the only thing I found that actually explains what getIn() does lol. I don't know why there is no documentation for it

File: public/index.html (hunk 1 above)
Comment (thaigillespie) [reply public/index.html:12]: Okay yeah it's indentation changes made automatically by VSCode, the actual code is still the same.

File: src/components/SignUpForm/styles.js (hunk 10 above)
Comment (thaigillespie) [reply src/components/SignUpForm/styles.js:3]: These bottom margins are there so that all the Formik's <Field /> are spaced out evenly from each other. Any other way to do it?

Comment (thaigillespie) [reply src/components/SignUpForm/styles.js:3]: nvm fixed it, eliminated most hardcoded margin and padding

File: src/pages/SignUpPageNew/styles.js (hunk 11 above)
Comment (thaigillespie) [reply src/pages/SignUpPageNew/styles.js:28]: I hardcoded the logo width and height because that's the true size of the logo. Honestly all the hardcoded margins and paddings are there so the whole page looks nice, since it's a little messed up if I take them out. I'm not sure how to balance things out nicely with <Card /> like how I can do with <Grid />, so I guess that's why there are more hardcoded margins and paddings here.

**Comment by godwinpang** [comment]
also looks like dropdowns import is wrong cuz of casing - build fails rn

**Comment by thaigillespie** [comment]
Yeah that was the thing that I was talking about in one of your comment threads. Not sure why it happens

File: src/pages/SignUpPageNew/index.js [hunk 12]
```diff
@@ -19,12 +19,21 @@ class SignUpPage extends React.Component {
...
   }
 
-  handleSubmit = (values, setSubmitting) => {
+  handleSubmit = async (values, setSubmitting) => {
     const signUpSubmission = {
       ...values,
     };
 
-    console.log(JSON.stringify(signUpSubmission));
+    const response = await fetch(SIGN_UP_URL, {
```
Comment (godwinpang) [inline src/pages/SignUpPageNew/index.js:27]: throw this behind something in services - frontend pages should not know anything about how we're signing up

File: src/services/auth.js [hunk 13]
```diff
@@ -1,6 +1,10 @@
 import * as firebase from 'firebase/app';
 import 'firebase/auth';
 
+import SIGNUP_ENDPOINT from '../constants/endpoints';
+
+const SIGN_UP_URL = `http://localhost:3001${SIGNUP_ENDPOINT}`;
```
Comment (godwinpang) [inline src/services/auth.js:6]: can you add a TODO to fetch base url based on env instead

Comment (thaigillespie) [reply src/services/auth.js:6]: I'll add that and then push it to newSignupPage then?

**Comment by godwinpang** [comment]
Is this good? can we merge

**Comment by thaigillespie** [comment]
Yeah it should be good now. I was just waiting for an approval from someone else.
